converter/internal/promtailconvert/promtailconvert.go

```diff
@@ -131,12 +131,33 @@ func appendScrapeConfig(
 	diags *diag.Diagnostics,
 	gctx *build.GlobalContext,
```

Comment:
I think that if we have been using more fields (e.g. more than 3?) from the global context, that would certainly make sense.

Currently I suspect this will be quite limited with just a few fields and adding a large struct with everything in it such as `cfg *promtailcfg.Config` may result in larger cognitive load when reading this IMO. Seeing that there are only a few fields in the GlobalContext, I can quickly see that only few things are considered global.

Reply:
> I like the builder concept a lot and I'm thinking about it some more. 

Thanks! One thing that builder is good for is a long pipeline where some steps are optional. I can have `discovery.* -> discovery.relabel -> local.file_match -> loki.source.file -> loki.process -> loki.write` and some of these intermediate components are optional. With the builder pattern I can skip a stage in the pipeline without too much code.